### ISSUE TITLE
Errors in the Capitalization of keys is not detected as an error

### ISSUE THREAD

**Issue by pmav99** [rhysd/actionlint#302]
actionlint does not seem to detect errors in the capitalization of keys. For example `run` vs `ruN`. The latter prevents the action from running altogether with the following error:
```
Error: .github#L1
every step must define a `uses` or `run` key
```
Nevertheless, actionlint says that there are no issues whatsoever.

Sample yaml file:

``` yaml
---

name: "Action Name"

on:
  push:

jobs:
  run_tests:
    name: "Test stacked environments"
    runs-on: "ubuntu-latest"

    steps:
      - ruN: echo "${PATH}"
```

**Comment by rhysd** [comment]
Some keys are case sensitive and some other keys are not. This needs more investigation.
